facade_charm/fetch_interfaces_from_charmhub.py

```diff
@@ -0,0 +1,61 @@
+"""Scrape charmhub to grab all interfaces for all registered charms.
+"""
+import asyncio
+import json
+import logging
+from pathlib import Path
+
+import aiohttp
+import requests
+import tenacity
+import yaml
+
+logger = logging.getLogger("update-endpoints")
+
+FACADE_CHARM_ROOT = Path(__file__).parent
+CH_INTERFACES_PATH = FACADE_CHARM_ROOT / 'charmhub_interfaces.yaml'
+
+
+def _get_all_registered_charms():
+    print('fetching store...')
+    resp = requests.get("https://charmhub.io/packages.json")
+    return resp.json()['packages']
+
+
+@tenacity.retry(stop=tenacity.stop_after_attempt(10), wait=tenacity.wait_fixed(1))
+async def get_endpoints(charm_name, session: aiohttp.ClientSession):
+    url = f"https://charmhub.io/{charm_name}/integrations.json"
+    async with session.get(url=url, timeout=4) as response:
+        raw = await response.read()
+        return json.loads(raw)['grouped_relations']
+
+
+async def get_all_integrations(charms_pkg_info):
+    async with aiohttp.ClientSession() as session:
```

Comment:
```suggestion
    async with aiohttp.ClientSession(raise_for_status=True) as session:
```

I would also include a base URL, it's pretty handy.